mautic/mautic
# Add 404 option in configuration

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | Y
| Automated tests included? | Y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
1. Add an option in Configuration --> System Settings --> General Settings (with the Root URL selection), enabling the user to select a landing page to use as the customized 404 page just as they would with the root URL
2. Use the same tooltip as for the Mautic's root URL field and just replace "Root URL" with "404"
3. If the 404 page is unpublished then use the default 404 page

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. N/A 

#### Steps to test this PR:
1. Go to Components -> Landing pages -> Add new landing page
2. Go to Configuration --> System Settings --> General Settings
3. There should be field **404 page** with published landing pages for selection
4. Select any landing page and save
5. Now selected landing page should be display instead of 404 error page
6. if 404 page not defined or selected landing page is unpublished then display default 404 error page.

#### Other areas of Mautic that may be affected by the change:
1. 
2. 

#### List deprecations along with the new alternative:
1. 
2. 

#### List backwards compatibility breaks:
1. 
2.
